apps/meteor/app/lib/server/functions/saveUserIdentity.ts

```diff
@@ -125,7 +126,8 @@ async function updateUsernameReferences({
 		const cursor = Messages.findByMention(previousUsername);
 		for await (const msg of cursor) {
 			const updatedMsg = msg.msg.replace(new RegExp(`@${previousUsername}`, 'ig'), `@${username}`);
-			await Messages.updateUsernameAndMessageOfMentionByIdAndOldUsername(msg._id, previousUsername, username, updatedMsg);
+			const updatedMd = messageTextToAstMarkdown(updatedMsg);
+			await Messages.updateUsernameAndMessageAndMdOfMentionByIdAndOldUsername(msg._id, previousUsername, username, updatedMsg, updatedMd);
```

Comment:
I think we should check the `DISABLE_MESSAGE_PARSER` env var here. Otherwise, we'd parse messages even though this has been disabled. Maybe we could just unset the `md` field when this env var is enabled (and keep what you did for when it's not).